okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java

```diff
@@ -197,7 +215,11 @@ public Level getLevel() {
 
         logger.log("");
         if (isPlaintext(buffer)) {
-          logger.log(buffer.readString(charset));
+          if (buffer.size() <= getBodyLogMax() || getBodyLogMax() < 0) {
+            logger.log(buffer.readString(charset));
+          } else {
+            logger.log(buffer.readString(getBodyLogMax(), charset));
```

Comment:
It is probably worth a comment that we still read the full body here, just display the first limit bytes, since memory pressure it still involves reading into a large array.  With an (public) API change this could be fixed, or alternatively by failing after limit bytes, catching and continuing.

Should we fix this also if the goal is "Recently I made a massive size of post request and my app killed by OOM."